src/main/java/org/apache/accumulo/testing/merkle/cli/GenerateHashes.java

```diff
@@ -151,82 +129,78 @@ public void setSplitsFile(String splitsFile) {
 
   public void run(GenerateHashesOpts opts) throws TableNotFoundException, AccumuloSecurityException, AccumuloException, NoSuchAlgorithmException,
       FileNotFoundException {
-    Collection<Range> ranges = getRanges(opts.getClient(), opts.getTableName(), opts.getSplitsFile());
-
-    run(opts.getClient(), opts.getTableName(), opts.getOutputTableName(), opts.getHashName(), opts.getNumThreads(), opts.isIteratorPushdown(), ranges);
+    try (AccumuloClient client = opts.createClient()) {
+      Collection<Range> ranges = getRanges(client, opts.getTableName(), opts.getSplitsFile());
+      run(client, opts.getTableName(), opts.getOutputTableName(), opts.getHashName(), opts.getNumThreads(), opts.isIteratorPushdown(), ranges);
+    }
   }
 
   public void run(final AccumuloClient client, final String inputTableName, final String outputTableName, final String digestName, int numThreads,
-      final boolean iteratorPushdown, final Collection<Range> ranges) throws TableNotFoundException, AccumuloSecurityException, AccumuloException,
+      final boolean iteratorPushdown, final Collection<Range> ranges) throws TableNotFoundException, AccumuloException,
       NoSuchAlgorithmException {
     if (!client.tableOperations().exists(outputTableName)) {
       throw new IllegalArgumentException(outputTableName + " does not exist, please create it");
     }
 
     // Get some parallelism
     ExecutorService svc = Executors.newFixedThreadPool(numThreads);
-    final BatchWriter bw = client.createBatchWriter(outputTableName, new BatchWriterConfig());
 
-    try {
+    try (BatchWriter bw = client.createBatchWriter(outputTableName, new BatchWriterConfig())) {
```

Comment:
Could this be `try (BatchWriter bw = client.createBatchWriter(outputTableName)) {`?